csrc/optimization/alias_analysis.h

```diff
@@ -29,27 +40,37 @@ class AliasAnalysisResult {
   AliasAnalysisResult(AliasAnalysisResult&&) = default;
   AliasAnalysisResult& operator=(AliasAnalysisResult&&) = default;
 
-  // Returns itself if `alias` doesn't alias anything.
-  const Val* findRoot(const Val* alias) const;
+  // Marks `source` as the immediate aliasing source of `alias` and sets the
+  // preferred layout.
+  void add(const TensorView* alias, const TensorView* source, Layout&& layout);
+
+  void finalize(Fusion* fusion);
 
   // Returns the preferred layout. If `alias` is not in `preferred_layout_`,
   // returns the `TensorView`'s initial layout.
   Layout preferredLayout(const Val* alias) const;
 
-  // Marks `source` as the immediate aliasing source of `alias` and sets the
-  // preferred layout.
-  void add(const TensorView* alias, const TensorView* source, Layout&& layout);
-
   std::string toString(int indent_size) const;
 
+  // Gets the root of the aliasing chain of a fusion output. Returns nullptr
+  // when `fusion_out` is not a fusion output or does not alias a fusion input.
+  const TensorView* getRoot(const TensorView* fusion_out) const;
```

Comment:
nitpick: `getRoot` and `findRoot` are a bit too close in name, yet differs in behavior regarding return `nullptr` & return self. Maybe this one could be renamed?

Reply:
Done.